hail/python/hail/expr/expressions/base_expression.py

```diff
@@ -946,7 +946,7 @@ def _to_relational_preserving_rows_and_cols(self, fallback_name):
             if self is source.col:
                 return None, source.key_cols_by().cols()
             if self is source.col_key:
-                return None, source.key_cols_by().cols().select()
```

Comment:
Won't this now produce a warning every time we call `_to_relational_...`, that seems bad. Can we somehow fix this without triggering the column table is sorted blah blah warning?

Reply:
initially disagreed, but this should produce an unkeyed thing the same as `col`. Fixed to `select_cols().key_cols_by().cols()`